src/commands/registries/azure/openInAzurePortal.ts

```diff
@@ -3,27 +3,28 @@
  *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { IActionContext } from '@microsoft/vscode-azext-utils';
+import { IActionContext, UserCancelledError, contextValueExperience } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
 import { ext } from '../../../extensionVariables';
-import { AzureRegistryTreeItem } from '../../../tree/registries/azure/AzureRegistryTreeItem';
-import { AzureRepositoryTreeItem } from '../../../tree/registries/azure/AzureRepositoryTreeItem';
-import type { SubscriptionTreeItem } from '../../../tree/registries/azure/SubscriptionTreeItem'; // These are only dev-time imports so don't need to be lazy
-import { registryExpectedContextValues } from '../../../tree/registries/registryContextValues';
-import { getAzExtAzureUtils, getAzSubTreeItem } from '../../../utils/lazyPackages';
+import { AzureRegistry, AzureSubscriptionRegistryItem, isAzureRegistry, isAzureSubscriptionRegistryItem } from '../../../tree/registries/Azure/AzureRegistryDataProvider';
+import { UnifiedRegistryItem } from '../../../tree/registries/UnifiedRegistryTreeDataProvider';
 
-export async function openInAzurePortal(context: IActionContext, node?: SubscriptionTreeItem | AzureRegistryTreeItem | AzureRepositoryTreeItem): Promise<void> {
+export async function openInAzurePortal(context: IActionContext, node?: UnifiedRegistryItem<AzureRegistry | AzureSubscriptionRegistryItem>): Promise<void> {
     if (!node) {
-        node = await ext.registriesTree.showTreeItemPicker<AzureRegistryTreeItem>(registryExpectedContextValues.azure.registry, context);
+        node = await contextValueExperience(context, ext.registriesTree, { include: ['azuresubscription', 'azureContainerRegistry'] });
     }
 
-    const azSubTreeItem = await getAzSubTreeItem();
-    const azExtAzureUtils = await getAzExtAzureUtils();
+    const azureRegistryItem = node.wrappedItem;
+    const baseUrl = `${azureRegistryItem.subscription.environment.portalUrl}/#@${azureRegistryItem.subscription.tenantId}/resource`;
+    let url: string;
 
-    if (node instanceof azSubTreeItem.SubscriptionTreeItem) {
-        await azExtAzureUtils.openInPortal(node.subscription, node.subscription.subscriptionId);
```

Comment:
Does `azextAzureUtils.openInPortal` only accept an `ISubscriptionContext`? If so I'd just convert the `AzureSubscription` to `ISubscriptionContext`, rather than trying to make a custom URL

Reply:
yeah you're right